### PR TITLE
add temperature.sapm_cell_from_module

### LINKED ISSUE #927: [ENH] cell temperature from backside of module temperature measurements
**Is your feature request related to a problem? Please describe.**
recently a question was posed, given backside module temperature, how to include them in the model chain? Looking at the temperature models available, there isn't one that takes module temperature as input and returns cell temperature. [SAPM has a formula for adjusting module temperature to obtain cell temperature](https://pvpmc.sandia.gov/modeling-steps/2-dc-module-iv/cell-temperature/sandia-cell-temperature-model/) based on irradiance and constant "delta-T" parameter that is characteristic of the module. However this formula is bundled into `pvlib.temperature.sapm_cell`as [Eq (2)](https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.temperature.sapm_cell.html#equation-sapm2 ) and the function doesn't take module temperature as input

**Describe the solution you'd like**
add an optional kwarg to `pvlib.temperature.sapm_cell` that allows it to bypass the module temperature calculation.

https://github.com/pvlib/pvlib-python/blob/ae98bf94413ac1495c8fe3ed4ce985a3f74c3805/pvlib/temperature.py#L117

##### option 1:
add `module_temperature` as kwarg
```python
def sapm_cell(poa_global, temp_air, wind_speed, a, b, deltaT,
              irrad_ref=1000, module_temperature=None):
    if module_temperature is None:
        module_temperature = sapm_module(poa_global, temp_air, wind_speed,
    else:
        warning.warn('air temp and wind speed parameters ignored, overridden by module temp')
        return module_temperature + (poa_global / irrad_ref) * deltaT
```

##### option 2:
add new `pvlib.temperature.sapm_module_to_cell` func
```python
def sapm_module_to_cell(module_temperature, poa_global, deltaT, irrad_ref=1000):
    return module_temperature + (poa_global / irrad_ref) * deltaT
```

_note_: if option 2, then `pvlib.temperature.sapm_cell` could be updated to call `pvlib.temperature.sapm_module_to_cell`

**Describe alternatives you've considered**
* We could explain to users that they can _back out_ the conditions to the use `pvlib.temperature.sapm_cell`
* we could encourage users to formulate the equation themselves from the literature
* we could implement a 3rd SAPM function to perform this task seperately, call it `pvlib.temperature.sapm_module_to_cell`?

**Additional context**
related to [this post on the Google group](https://groups.google.com/d/msg/pvlib-python/CBThHDydaxs/oYuaKwJmAQAJ)

### PR DESCRIPTION
**Pull request by cwhanse**
- [x] Closes #927
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - [x] Updates entries to [`docs/sphinx/source/api.rst`](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/api.rst) for API changes.
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

### CODE REVIEW

File: pvlib/temperature.py [hunk 1]
```diff
@@ -186,6 +187,72 @@ def sapm_module(poa_global, temp_air, wind_speed, a, b):
...
+    +===============+================+=======+=========+=====================+
+    | glass/glass   | open rack      | -3.47 | -0.0594 | 3                   |
+    +---------------+----------------+-------+---------+---------------------+
+    | glass/glass   | close roof     | -2.98 | -0.0471 | 1                   |
+    +---------------+----------------+-------+---------+---------------------+
+    | glass/polymer | open rack      | -3.56 | -0.075  | 3                   |
+    +---------------+----------------+-------+---------+---------------------+
+    | glass/polymer | insulated back | -2.81 | -0.0455 | 0                   |
+    +---------------+----------------+-------+---------+---------------------+
+
```
Comment (mikofski) [inline pvlib/temperature.py]: How about adding a See Also section here to link to `sapm_cell` or other relevant calls

```

    See also
    --------
    sapm_cell
```

File: pvlib/temperature.py [hunk 2]
```diff
@@ -186,6 +187,72 @@ def sapm_module(poa_global, temp_air, wind_speed, a, b):
...
+       T_{C} = T_{m} + \frac{E}{E_{0}} \Delta T
+
+    The module back surface temperature :math:`T_{m}` is implemented in
+    :py:func:`~pvlib.temperature.sapm_module`.
+
+    Model parameters depend both on
+    the module construction and its mounting. Parameter sets are provided in
+    [1]_ for representative modules and mounting, and are coded for convenience
+    in ``pvlib.temperature.TEMPERATURE_MODEL_PARAMETERS``.
```
Comment (mikofski) [inline pvlib/temperature.py]: The constant dictionary `TEMPERATURE_MODEL_PARAMETERS` is not included in the public api (`api.rst`) even though the documentation references it numerous places. If it's worth mentioning to users then IMO I think it should be published. Perhaps a separate bug/issue?

File: pvlib/temperature.py [hunk 3]
```diff
@@ -33,7 +33,7 @@ def _temperature_model_params(model, parameter_set):
 def sapm_cell(poa_global, temp_air, wind_speed, a, b, deltaT,
               irrad_ref=1000):
     r'''
-    Calculate cell temperature per the Sandia PV Array Performance Model.
+    Calculate cell temperature per the Sandia  Array Performance Model.
```
Comment (wholmgren) [inline pvlib/temperature.py]:
```suggestion
    Calculate cell temperature per the Sandia Array Performance Model.
```

File: pvlib/temperature.py [hunk 4]
```diff
@@ -182,10 +188,84 @@ def sapm_module(poa_global, temp_air, wind_speed, a, b):
...
+
+       T_{C} = T_{m} + \frac{E}{E_{0}} \Delta T
+
+    The module back surface temperature :math:`T_{m}` is implemented in
+    :py:func:`~pvlib.temperature.sapm_module`.
+
+    Model parameters depend both on
+    the module construction and its mounting. Parameter sets are provided in
+    [1]_ for representative modules and mounting, and are coded for convenience
+    in ``pvlib.temperature.TEMPERATURE_MODEL_PARAMETERS``.
```
Comment (wholmgren) [inline pvlib/temperature.py]: nit: fix line breaks

File: pvlib/temperature.py [hunk 5]
```diff
@@ -182,10 +188,84 @@ def sapm_module(poa_global, temp_air, wind_speed, a, b):
...
+    | Module        | Mounting       | a     | b       | :math:`\Delta T [C]`|
+    +===============+================+=======+=========+=====================+
+    | glass/glass   | open rack      | -3.47 | -0.0594 | 3                   |
+    +---------------+----------------+-------+---------+---------------------+
+    | glass/glass   | close roof     | -2.98 | -0.0471 | 1                   |
+    +---------------+----------------+-------+---------+---------------------+
+    | glass/polymer | open rack      | -3.56 | -0.075  | 3                   |
+    +---------------+----------------+-------+---------+---------------------+
+    | glass/polymer | insulated back | -2.81 | -0.0455 | 0                   |
+    +---------------+----------------+-------+---------+---------------------+
```
Comment (wholmgren) [inline pvlib/temperature.py]: I don't like that this table is copy/pasted across multiple functions. I'm ok with it for now, but perhaps resolving #939 will allow us to remove it from both docstrings.

**Comment by CameronTStark** [comment]
Just FYI, I re-ran the two failing conda_windows tests and they came back passing. 

I did so while comparing CI failures in PRs to compare to the ones that popped up for #937 (spoiler: they were different).

File: docs/sphinx/source/whatsnew/v0.7.2.rst [hunk 6]
```diff
@@ -20,8 +20,10 @@ Enhancements
   ``temperature_model='faiman'`` option to :py:class:`~pvlib.modelchain.ModelChain`
   (:pull:`897`) (:issue:`836`).
 * Add Kimber soiling model :py:func:`pvlib.losses.soiling_kimber`. (:pull:`860`)
-* Add :func:`~pvlib.iotools.read_pvgis_tmy` for files downloaded using the
+* Add :py:func:`~pvlib.iotools.read_pvgis_tmy` for files downloaded using the
   PVGIS tool. (:issue:`880`)
+* Add :py:func:`~pvlib.temperature.sapm_cell_from_module` to convert back-of-
+  module temperature to cell temperature (:issue:`927`)
```
Comment (wholmgren) [inline docs/sphinx/source/whatsnew/v0.7.2.rst:26]: remove `~` so that sphinx renders the full path.

**Comment by CameronTStark** [comment]
Sorry, I merged #937 which caused another conflict here. If you don't mind re-un-conflicting @cwhanse.

**Comment by CameronTStark** [comment]
Thank you @cwhanse!

**Pull request merged**